torch/utils/_sympy/interp.py

```diff
@@ -138,6 +138,12 @@ def _run_sympy_handler(analysis, args, expr, index_dtype=torch.int64):
     if (handler_name := INDEX_DTYPE_HANDLERS.get(expr.func)) is not None:
         return getattr(analysis, handler_name)(*args, index_dtype)
 
+    # Fastpath for n-ary integral addition
+    if expr.func is sympy.Add and expr.is_integer and hasattr(analysis, "sym_add"):
```

Comment:
Do we need to consider `is_integer`? It can be expensive.

Reply:
There's some other code that is simpler if I assume the n-ary only happens for ints. I don't have to make this assumption, but then I need to make sure I do float promotion correctly.

This is all kind of annoying because is_integer should be cheap; I'm really using is_integer as a stand in for "would I infer integer under some simple static typing rules that don't involve complicated questions about assumptions" lol

Reply:
https://github.com/isuruf/pytorch/commit/b30aa11a885a10df435f564f1071fd14593e141f was my attempt at this last week, but it didn't help.

Reply:
Ah, very similar, I am sorry I did not see it before doing mine lol. I wonder if need to find more spots where operator.add is being called and short circuit it faster.